### ISSUE TITLE
local methods of RawMultipurpose & Analysis

### ISSUE THREAD

**Issue by qTich** [renpy/renpy#5832]
`check_spline_types` of `RawMultipurpose`
`check_name`, `check_nodes` & `check_node` of `Analysis` 

Because the methods are local and used once, they create a lot of garbage (gc almost always clears them (`config.manage_gc = config.gc_print_unreachable = True`)), and the bigger the game itself, the worse they make things worse. 

They should be taken out for access to the global area.

**Comment by qTich** [comment]
Also, if I understand correctly, when a language other than `None` is selected, each cleanup includes - `group_logical_lines`, `depth_split`, `gll_core` and the `Lexer(nested)` block itself.

**Comment by renpytom** [comment]
Yes, I'm using static analysis to find the more obvious ones. At the same time, I'm limiting it to ones that matter for performance - I'm fine with there being GC on rare events like a language change. (But I fixed this, since I suspect it will boost load time.)
